src/application_settings/container_base.py

```diff
@@ -138,25 +138,28 @@ def _save(self) -> Self:
         return self
 
     @classmethod
-    def _get_saved_data(cls) -> dict[str, Any]:
+    def _get_saved_data(cls, throw_if_file_not_found: bool = False) -> dict[str, Any]:
         """Get the data stored in the parameter file"""
         data_stored: dict[str, Any] = {}
 
         if path := cls.filepath():
-            if (ext := path.suffix[1:].lower()) == str(FileFormat.TOML.value):
-                with path.open(mode="rb") as fptr:
-                    data_stored = tomllib.load(fptr)
-            elif ext == str(FileFormat.JSON.value):
-                with path.open(mode="r") as fptr:
-                    data_stored = json.load(fptr)
-            else:
-                print(f"Unknown file format {ext} given in {path}.")
+            if throw_if_file_not_found or path.is_file():
```

Comment:
I'm not such a fan of letting `tomllib.load` or `json.load` throw the file not exist error, feels like we're a bit too dependent on their implementation.

I think something like:

```python
if not path.is_file() and throw_if_file_not_found:
    raise FileNotFoundError("some explanation")

if (ext := ....)
```

Reply:
We could even move this more to the top to avoid the first if/else:

```python
path = cls.filepath()

if path is None or not path.is_file():
    err_mess = f"No path specified for {cls.kind_string().lower()} file."
    if raise_if_file_not_found:
        raise FileNotFoundError(err_mess )
    print(err_mess, "Trying with defaults, but this may not work.")
    return {}

data_stored = {}
if (ext := path...
```

Reply:
Agreed, done.